core/types/transaction.go

```diff
@@ -58,9 +65,10 @@ type txdata struct {
 	S *big.Int `json:"s" gencodec:"required"`
 
 	// This is only used when marshaling to JSON.
-	Hash            *common.Hash    `json:"hash" rlp:"-"`
-	L1RollupTxId    *hexutil.Uint64 `json:"l1RollupTxId,omitempty" rlp:"nil,?"`
-	L1MessageSender *common.Address `json:"l1MessageSender,omitempty" rlp:"nil,?"`
+	Hash              *common.Hash       `json:"hash" rlp:"-"`
+	L1RollupTxId      *hexutil.Uint64    `json:"l1RollupTxId,omitempty" rlp:"nil,?"`
+	L1MessageSender   *common.Address    `json:"l1MessageSender,omitempty" rlp:"nil,?"`
+	SignatureHashType *SignatureHashType `json:"signatureHashType,omitempty" rlp:"nil,?"`
```

Comment:
Ah this is a great point. I imagine that unless we pipe txs through postgres then txs with a `SignatureHashType` will not have a `L1RollupId `.

For this reason, it might make sense to add `SignatureHashType` before `L1RollupTxId`

Reply:
This could be a good time to create a new struct that contains all this rollup metadata. Maybe `RollupMetadata` or something? The idea is that then we can omit the entire struct from hashing logic while always having the contained properties present (even on nil), so we don't run into these "is it present or not?" issues in serlialization.

Reply:
I have moved each of `L1RollupTxId`, `L1MessageSender` and `SignatureHashType` to the `tx.meta` struct and they are no longer RLP encoded, so we don't have to worry about this problem. I altered the tests around checking for the RLP serialization to be different when the metadata is included to test that the RLP serialization is the same when the metadata is included.